test/active_record_doctor/detectors/unindexed_foreign_keys_test.rb

```diff
@@ -14,6 +14,34 @@ def test_unindexed_foreign_key_is_reported
     OUTPUT
   end
 
+  def test_unindexed_polymorphic_foreign_key_is_reported
+    skip("MySQL always indexes foreign keys") if mysql?
```

Comment:
It seems this restriction can be removed, as we're not dealing with real foreign keys.